godot-bindings/src/godot_json.rs

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) godot-rust; Bromeon and contributors.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+//! Commands related to parsing user-provided JSON and extension headers.
+
+use crate::depend_on_custom_json::header_gen::{generate_rust_binding, patch_c_header};
+use crate::{GodotVersion, StopWatch};
+use std::fs;
+use std::path::Path;
+
+use crate::godot_version::validate_godot_version;
+use nanoserde::DeJson;
+
+// At first re-using mapping from godot-codegen json.rs might seem more desirable but there are few issues to consider:
+// * Overall JSON file structure might change slightly from version to version, while header should stay consistent (otherwise it defeats the purpose of having any header at all).
+// Having two parsers – minimal one inherent to api-custom-json feature and codegen one – makes handling all the edge cases easier.
+// * `godot-codegen` depends on `godot-bindings` thus simple re-using types from former in side the latter is not possible (cyclic dependency).
+// Moving said types to `godot-bindings` would increase the cognitive overhead (since domain mapping is responsibility of `godot-codegen`, while godot-bindings is responsible for providing required resources & emitting the version).
+// In the future we might experiment with splitting said types into separate crates.
```

Comment:
This should come at the beginning of the file, after `//!` (can still stay in separate `//` paragraph, just before imports).